SWLOR.Game.Server/Native/GetDamageRoll.cs

```diff
@@ -34,38 +37,85 @@ private static int OnGetDamageRoll(void* thisPtr, void* pTarget, int bOffHand, i
             var targetObject = CNWSObject.FromPointer(pTarget);
             var damageFlags = attackerStats.m_pBaseCreature.GetDamageFlags();
 
-            var dmg = 0f;
-            var attackAttribute = attackerStats.m_nStrengthBase < 10 ? 0 : attackerStats.m_nStrengthModifier;
-            var damage = 0;
-            var specializationDMGBonus = 0f;
-            Log.Write(LogGroup.Attack, "DAMAGE: attacker strength mod: " + attackAttribute.ToString());
-            // TODO - ranged attacks (and other feat based damage) should not add the strength score.
+            // On a critical hit, this method appears to be invoked multiple times, with an invalid target the second and
+            // subsequent times.  Bail out early.
+            if (targetObject == null) return 0;
 
-            // Calculate attacker's DMG
-            if (attacker != null)
-            {
-                var weapon = bOffHand == 1
+            uint attackType = (uint) AttackType.Melee;
+            // Developer note - the native functions to retrieve a combat round don't work at this point.
+            CNWSItem weapon = bOffHand == 1
```

Comment:
General nitpick: I prefer to use var instead of specific types. Helps with refactoring later. I have a tool that'll automatically change these so it's not a big deal if you don't want to do it now.

Reply:
Yeah, I was always taught to be specific about types, was drilled into me when I was learning to code as making code easier for others to understand (unclear how true you find that, but that was how I was trained :)).  So it's a hard habit for me to break :)

Reply:
Haha same here, but I've found it fairly beneficial for the types to be determined by the compiler. On occasion I'll be explicit if it's not clear but in general VS / Resharper works well. Again, just nitpicking - I can live with it either way.